File: 0A-binary-tree/main.py
from BinaryTree import BinaryTree

# create a new binary tree
tree = BinaryTree()

# insert elements into the tree
tree.Insert(1)
tree.Insert(2)
tree.Insert(3)
tree.Insert(4)
tree.Insert(5)
tree.Insert(6)
tree.Insert(7)

# print the height of the tree
print("Height: ", tree.height())

# traverse the tree in pre-order
tree.PreOrder()

# traverse the tree in in-order
tree.InOrder()

# traverse the tree in post-order
tree.PostOrder()

File: 0A-binary-tree/main.cs
using System;
using metwally;

class Program
{
    public static void Main(string[] args)
    {
        BinaryTree<char> tree = new BinaryTree<char>();
        tree.Insert('A');
        tree.Insert('B');
        tree.Insert('C');
        tree.Insert('D');
        tree.Insert('E');
        tree.Insert('F');
        tree.Insert('G');
        tree.Insert('H');
        tree.Insert('I');
        tree.Print();
        
        Console.WriteLine("Height: " + tree.Height());
        tree.PreOrder();
        tree.InOrder();
        tree.PostOrder();

    }
}
